filters/LiTreeFilter.cpp

```diff
@@ -162,29 +162,30 @@ void LiTreeFilter::classifyPoint(PointId ui, PointView& view, PointIdList& Ni,
     PointRef u = view.point(ui);
     double Zu = u.getFieldAs<double>(Id::HeightAboveGround);
 
-    PointViewPtr Nview = view.makeNew();
-    for (PointId const& n : Ni)
-        Nview->appendPoint(view, n);
-    KD2Index& Ntree = Nview->build2dIndex();
+    // compute dmin1 and dmin2 (nearest neighbor in each set)
+    auto mindist = [&ui, &view](PointIdList idx) {
+        double ux = view.getFieldAs<double>(Id::X, ui);
+        double uy = view.getFieldAs<double>(Id::Y, ui);
 
-    PointViewPtr Pview = view.makeNew();
-    for (PointId const& p : Pi)
-        Pview->appendPoint(view, p);
-    KD2Index& Ptree = Pview->build2dIndex();
+        std::vector<double> dists;
+        for (PointId const& i : idx)
```

Comment:
I must be missing something here, but I don't understand why we're building a list and then extracting the minimum value, rather than just checking the current `dist` against the minimum that we've found so far.

Reply:
You aren't missing anything per se. It's just the way I happened to implement it. I can switch.